File: Problems/Generic/Create Tree from Traversals/Inorder & Level/tree.js
class Node {
    constructor(data, left = null, right = null) {
        this.data = data
        this.left = left
        this.right = right
    }

    inorder() {
        function inorder(node) {
            if (node == null) {
                return ''
            }

            return inorder(node.left)
                + node.data + ' '
                + inorder(node.right)
        }
        return inorder(this)
    }

    levelorder() {
        function levelorder(root) {
            let output = ''

            let queue = [root]
            while (queue.length > 0) {
                let node = queue.shift()
                output += node.data + ' '

                if (node.left) {
                    queue.push(node.left)
                }
                if (node.right) {
                    queue.push(node.right)
                }
            }
            return output
        }
        return levelorder(this)
    }
}

export { Node }

File: problems/Generic/Tree From Parent/tree.py
class Node:
    def __init__(self, data, left=None, right=None):
        self.data = data

        self.left = left
        self.right = right

    def __str__(self):
        def preorder(node):
            if not node:
                return ''
            return str(node.data) + ' ' + preorder(node.left) + preorder(node.right)

        return preorder(self)